File: 3rd-semester/object-oriented-programming/1st-assignment/csharp/src/Models/Sms.cs
namespace SurveillanceRegistry.Models;

/// <summary>Represents an SMS text message communication.</summary>
public sealed class Sms : Communication
{
    public string Content { get; }

    public Sms(string senderNumber, string receiverNumber,
               int day, int month, int year, string content)
        : base(senderNumber, receiverNumber, day, month, year)
        => Content = content;

    public override int    GetDuration()   => 0;
    public override string GetSmsContent() => Content;

    public override string ToString() =>
        $"SMS[{base.ToString()}, content=\"{Content}\"]";
}

File: 3rd-semester/object-oriented-programming/2nd-assignment/java/src/main/java/com/university/oop/surveillance/model/Sms.java
package com.university.oop.surveillance.model;

import java.util.logging.Logger;

/** Represents an SMS text message communication. */
public final class Sms extends Communication {

    private static final Logger LOGGER = Logger.getLogger(Sms.class.getName());
    private final String content;

    public Sms(String senderNumber, String receiverNumber,
               int day, int month, int year, String content) {
        super(senderNumber, receiverNumber, day, month, year);
        this.content = content;
    }

    @Override public int getDuration()      { return 0; }
    @Override public String getSmsContent() { return content; }

    @Override
    public void printInfo() {
        LOGGER.info("SMS details:");
        super.printInfo();
        LOGGER.info("Text: " + content);
    }

    @Override
    public String toString() {
        return String.format("SMS[%s, content=\"%s\"]", super.toString(), content);
    }
}
